ui/src/adapters/api/identities.ts

```diff
@@ -0,0 +1,193 @@
+        Authorization: buildAuthorizationHeader(env),
+      },
+      method: "GET",
+      signal,
+      url: `${API_VERSION}/identities`,
+    });
+
+    return buildSuccessResponse(getListParser(identityParser).parse(response.data || []));
+  } catch (error) {
+    return buildErrorResponse(error);
+  }
+}
+
+export type CreateIdentity = {
+  blockchain: string;
+  credentialStatusType: CredentialStatusType;
+  displayName: string;
+  method: string;
+  network: string;
+  type: IdentityType;
+};
+
+type CreatedIdentityResponse = {
+  identifier: string;
+};
+
+export const createIdentityResponseParser = getStrictParser<CreatedIdentityResponse>()(
+  z.object({
+    identifier: z.string(),
+  })
+);
+
+export async function createIdentity({
+  env,
+  payload,
+}: {
+  env: Env;
+  payload: CreateIdentity;
+}): Promise<Response<CreatedIdentityResponse>> {
+  try {
+    const { credentialStatusType, displayName, ...didMetadata } = payload;
+    const response = await axios({
+      baseURL: env.api.url,
+      data: { credentialStatusType, didMetadata, displayName },
+      headers: {
+        Authorization: buildAuthorizationHeader(env),
+      },
+      method: "POST",
+      url: `${API_VERSION}/identities`,
+    });
+
+    return buildSuccessResponse(createIdentityResponseParser.parse(response.data));
+  } catch (error) {
+    return buildErrorResponse(error);
+  }
+}
+
+export const identityDetailsParser = getStrictParser<IdentityDetails>()(
+  z.object({
+    credentialStatusType: z.nativeEnum(CredentialStatusType),
+    displayName: z.string(),
+    identifier: z.string(),
+    keyType: z.nativeEnum(IdentityType),
+  })
+);
+
+export async function getIdentity({
+  env,
+  identifier,
+  signal,
+}: {
+  env: Env;
+  identifier: Identifier;
+  signal?: AbortSignal;
+}): Promise<Response<IdentityDetails>> {
+  try {
+    const response = await axios({
+      baseURL: env.api.url,
+
+      headers: {
+        Authorization: buildAuthorizationHeader(env),
+      },
+      method: "GET",
+      signal,
+      url: `${API_VERSION}/identities/${identifier}/details`,
+    });
+
+    return buildSuccessResponse(identityDetailsParser.parse(response.data));
+  } catch (error) {
+    return buildErrorResponse(error);
+  }
+}
+
+export async function updateIdentityDisplayName({
+  displayName,
+  env,
+  identifier,
+}: {
+  displayName: string;
+  env: Env;
+  identifier: Identifier;
+}): Promise<Response<void>> {
+  try {
+    await axios({
+      baseURL: env.api.url,
+      data: { displayName },
+      headers: {
+        Authorization: buildAuthorizationHeader(env),
+      },
+      method: "PATCH",
+      url: `${API_VERSION}/identities/${identifier}`,
+    });
+
+    return buildSuccessResponse(undefined);
+  } catch (error) {
+    return buildErrorResponse(error);
+  }
+}
+
+export const supportedNetworkParser = getStrictParser<SupportedNetwork>()(
+  z.object({
+    blockchain: z.string(),
+    networks: z.array(z.string()).nonempty(),
```

Comment:
This parser better represents the type:

`z.tuple([z.string()]).rest(z.string())`